test/JDBC/expected/todatetimeoffset.out

```diff
@@ -0,0 +1,567 @@
+Select todatetimeoffset('0000-04-22 16:23:51.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('9999-04-22 16:23:51.766890','+12:00')
+GO
+~~START~~
+datetimeoffset
+9999-04-22 16:23:51.7668900 +12:00
+~~END~~
+
+
+Select todatetimeoffset('-1-04-22 16:23:51.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-0a-22 16:23:51.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('20sb-04-22 16:23:51.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-04-2c 16:23:51.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-04-22 1d:23:51.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-0a-22 16:2a:51.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-04-22 16:23:5d.766890','+12:00')
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Conversion failed when converting date and/or time from character string.)~~
+
+
+Select todatetimeoffset('2000-04-22 16:23:51.7668c0','+12:00')
```

Comment:
is this expected, not to fail for 7668c0?

Reply:
Yeah this is due to issue in the cast for datetime2 and datetimeoffset function.
```
1> Select cast('2000-04-22 16:23:51.7668c0' as datetime2)
2> go
datetime2                             
--------------------------------------
           2000-04-22 16:23:51.7668000

(1 rows affected)
1> Select cast('2000-04-22 16:23:51.7668c0' as datetimeoffset)
2> go
datetimeoffset                               
---------------------------------------------
           2000-04-22 16:23:51.7668000 +00:00
```

Reply:
you mean this is not expected but happens due to cast issue for dateime2/datetimeoffset. If that is the case, please add a comment with corresponding Jira.

Reply:
Addressed.